x-pack/plugins/lens/public/native_renderer/native_renderer.tsx

```diff
@@ -19,11 +25,42 @@ export interface NativeRendererProps<T> extends HTMLAttributes<HTMLDivElement> {
  * By default the mountpoint element will be a div, this can be changed with the
  * `tag` prop.
  *
+ * If the rendered component tree was using React, we need to clean it up manually,
+ * otherwise the unmount event never happens. A future addition is for non-React components
+ * to get cleaned up, which could be added in the future.
+ *
  * @param props
  */
 export function NativeRenderer<T>({ render, nativeProps, tag, ...rest }: NativeRendererProps<T>) {
+  const elementRef = useRef<Element>();
+  const cleanupRef = useRef<((cleanupElement: Element) => void) | void>();
+  useEffect(() => {
+    return () => {
+      if (elementRef.current) {
+        if (cleanupRef.current && typeof cleanupRef.current === 'function') {
+          cleanupRef.current(elementRef.current);
+        }
+        unmountComponentAtNode(elementRef.current);
```

Comment:
This PR adds a way to return a cleanup function (even an async one) from the render function, but it seems like this is never used, right?

The
```ts
unmountComponentAtNode(elementRef.current);
```
line is basically a default implementation of the cleanup specific to react and it does all the actual cleanup here.

I guess my question is: Do we actually need the unmount callback logic? Or is it more of a future-facing thing?

Reply:
Yes, the `cleanupRef` is only used in the new tests It's there to maintain the existing assumption that Lens is framework independent.